### ISSUE TITLE
Abortion and Paid Family Leave figure

### ISSUE THREAD

**Issue by alinasmahl1** [Drexel-UHC/analytics-corner#7]
### Name

Alina Schnake-Mahl

### Title

Abortion and Paid Family Leave Figure

### Department

Urban Health Collaborative

### Request type

Bespoke visualization 

### Request description

I'm hoping to make a figure that shows different policy statuses of states (abortion ban, proposed abortion ban, no abortion ban; Paid family leave, no paid family leave; paid family leave preemption, no preemption), with the states in the location of the US map...

I think geofacet mightttt get us close to the figure I'm hoping to make, but not quitee there. 

### Example

The first figure in this article https://www.scientificamerican.com/article/graphic-many-states-that-restrict-or-ban-abortion-dont-teach-kids-about-sex-and-pregnancy/ is what I'm thinking of. 

### Data

sent via slack (let me know if I should resend) 

### Notes

timeline ~ 2 weeks.

**Comment by ran-codes** [comment]
# Proof of Concept

@alinasmahl1 I made a POC [here](https://drexel-uhc.github.io/analytics-core/docs/blog/issues/7%20-%20Abortion%20and%20Paid%20Family%20Leave%20figure/quarto.html). Let me know what you think and how you want to proceed!

# Access Code

### On the blog post
![image](https://user-images.githubusercontent.com/45013044/184143745-5f6310a0-98eb-404f-8ab5-1d842dcc7ce5.png)

### Fork this repository 
The folder for this issue can be found [here](https://github.com/Drexel-UHC/analytics-core/tree/main/docs/blog/issues/7%20-%20Abortion%20and%20Paid%20Family%20Leave%20figure)

**Comment by alinasmahl1** [comment]
This looks fantastic, thank you @ran-codes!!! 

I updated the dataset-- would it be possible to use the new attached data? (I can also probably adapt the code to the new data, if that's easier!). But with the exception of updating the dataset,  I think basically where we want to go with it for now! I think I should plan to show it to my co-authors, and then if I can't figure out how to make changes they request, then I'll follow back up with you?

**Comment by ran-codes** [comment]
>I updated the dataset-- would it be possible to use the new attached data? 

Yes send the code via slack or upload in this issue. I can update the figure with new data.

**Comment by alinasmahl1** [comment]
This is so great, thank you!! really appreciate how you walk through what the code is doing!

Tiny thing, can we update the text about the preemption to say "Red triangles show state preemption of paid family policies 

yes, the color rectangles should only have 3 values, but I'm wondering if we can make it ordinal, so uses it the terms "accessible, hostile, and illegal" ? rather than using a risk scale. 

(also, just let me know if you'd rather I make these small edits-- I'm sure I can adapt the code since you've got it set up!)

**Comment by ran-codes** [comment]
your too kind 😊 !

made some quick changes that can be found [here](https://drexel-uhc.github.io/analytics-corner/docs/blog/issues/7%20-%20Abortion%20and%20Paid%20Family%20Leave%20figure/quarto.html)

**Comment by alinasmahl1** [comment]
@ran-codes I was really hoping I could update this myself, but I got a bit stuck.  The article co-authors asked that we include another category in the abortion variable, and wondering if you can help me update the figure? I slacked you the updated data

**Comment by ran-codes** [comment]
sure thing

## Some thoughts

###   The current viz has three data fields with represented by three visual layers (as shown below).

![image](https://user-images.githubusercontent.com/45013044/190727742-8dcda600-31e5-4a61-8131-61bcaa0d61ab.png)

### Proposed Update 

To add another data field (**Preempt + hositle/illegal**?) we would need to add another visual layer to represent this new field.   The original example also had 4 data fields and 4 visual layers (rectangle shading, blue triangle, orange triangle and the **outline (starred below)**.

![image](https://user-images.githubusercontent.com/45013044/190728539-3d10b228-aa56-48e5-a68e-2899a5aa5c6f.png)


# Questions 

So just a few things to confirm with you:
- Confirm that the data field we are adding is `Preempt + hositle/illegal`?
- Do you want to representing this new field visually with **outline (as circled above)?
- If you do what should the legend look like (title? label for the outlined one? label for the unlabeled one?)

**Comment by alinasmahl1** [comment]
@ran-codes  ah sorry, I think I communicated this confusingly- we just want to add an additional category to the abortion attitude legend (the grey)-- by splitting up the existing accessible section into into "accessible, protected" and "accessible, not protected". So color wise it would go "accessible, protected"- lightest color, "accessible, not protected" next lightest, then hostile, and then illegal. 

So I think we shouldn't add an additional field.

**Comment by ran-codes** [comment]
OOO. Got it !! Thats much easier 😄 

Will have something to you today.

**Comment by ran-codes** [comment]
How does this look?

![issue5](https://user-images.githubusercontent.com/45013044/191041354-a068c88a-81de-49e3-9fdf-1b95b728f590.jpeg)

**Comment by usamabilal** [comment]
One suggestion: it's a bit hard to differentiate accessible not protected from hostile in the legend, but it's easy in the figure: the color of the state text. (e.g. NM vs AZ).  (it's also tricky because there's only three states with accessible not protected, NM, VA, NH, right?)
A suggestion would be to include this specifically in the legend, by increasing the size of the rectangle, and including the text inside the rectangle following the same color as in the figure itself. does that make sense?

(just a suggestion, feel free to ignore if you think this makes things too complicated!)

**Comment by alinasmahl1** [comment]
o yup, love this!! Great suggestion-- I think this makes it much easier to see the distinction

**Comment by ran-codes** [comment]
@alinasmahl1 see the zipped bundle ([output-bundle-issue7.zip](https://github.com/Drexel-UHC/analytics-corner/files/9600084/output-bundle-issue7.zip)) which is a self contained project containing:
- index.R: R script to run to generate output
- the data used 
- the jpeg output


 (still experimenting with what type of output is best to give to people using the Analytics Corner - so let me know what format you would like the deliverables in).

**Comment by usamabilal** [comment]
> (still experimenting with what type of output is best to give to people using the Analytics Corner - so let me know what format you would like the deliverables in).

maybe include a radius question in the request form?
- Final output (+ open ended field with format of final output; e.g., jpg, png...)
- Reproducible package (data + code)

**Comment by alinasmahl1** [comment]
That output is great for me -- thank you again!!!
And yup, I like that idea @usamabilal --
